src/v/s3/client.cc

```diff
@@ -497,20 +498,28 @@ ss::future<http::client::response_stream_ref> client::get_object(
       "send https request:\n{}",
       http::redacted_header(header.value()));
     return _client.request(std::move(header.value()), timeout)
-      .then([](http::client::response_stream_ref&& ref) {
+      .then([expect_no_such_key](http::client::response_stream_ref&& ref) {
           // here we didn't receive any bytes from the socket and
           // ref->is_header_done() is 'false', we need to prefetch
           // the header first
-          return ref->prefetch_headers().then([ref = std::move(ref)]() mutable {
+          return ref->prefetch_headers().then([ref = std::move(ref),
+                                               expect_no_such_key]() mutable {
               vassert(ref->is_header_done(), "Header is not received");
               if (
                 ref->get_headers().result() != boost::beast::http::status::ok) {
                   // Got error response, consume the response body and produce
                   // rest api error
-                  vlog(
-                    s3_log.warn,
-                    "S3 replied with error: {}",
-                    ref->get_headers());
+                  if (expect_no_such_key) {
```

Comment:
makes sense, but shouldn't we only apply this logic if the error is really indicating that the key is not found?